controllers/podsweep_runner_test.go

```diff
@@ -132,10 +202,81 @@ var _ = Describe("PodSweeper runner", func() {
 						},
 					},
 				},
+				"",
+			},
+			{
+				"without annotation and with API that return future time",
+				corev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "sample4",
+						Namespace: namespace,
+						Labels: map[string]string{
+							constants.RunnerOrgLabelKey: organizationName,
+						},
+					},
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{
+								Name:  "sample",
+								Image: "sample:latest",
+							},
+						},
+					},
+				},
+				time.Now().Add(24 * time.Hour).Format(time.RFC3339),
+			},
+			{
+				"with future time annotation and without API",
+				corev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "sample5",
+						Namespace: namespace,
+						Labels: map[string]string{
+							constants.RunnerOrgLabelKey: organizationName,
+						},
+						Annotations: map[string]string{
+							constants.PodDeletionTimeKey: time.Now().Add(24 * time.Hour).UTC().Format(time.RFC3339),
+						},
+					},
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{
+								Name:  "sample",
+								Image: "sample:latest",
+							},
+						},
+					},
+				},
+				"",
+			},
+			{
+				"with future time annotation and with API that return time already past",
+				corev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "sample6",
+						Namespace: namespace,
+						Labels: map[string]string{
+							constants.RunnerOrgLabelKey: organizationName,
+						},
+						Annotations: map[string]string{
+							constants.PodDeletionTimeKey: time.Now().Add(24 * time.Hour).UTC().Format(time.RFC3339),
+						},
+					},
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{
+								Name:  "sample",
+								Image: "sample:latest",
+							},
+						},
+					},
+				},
+				time.Now().Add(24 * time.Hour).UTC().Format(time.RFC3339),
```

Comment:
It is a future time.

```suggestion
				time.Now().Add(24 * time.Hour).UTC().Format(time.RFC3339),
```